### PR TITLE
Add first draft of contribution_guide.rst

### PR DESCRIPTION
**Pull request by saroad2**

### CODE REVIEW

File: docs/contribution_guide.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+Contributors are the living, beating heart of any open-source project, and we really
+appreciate each and every contribution made to our code-base, big or small.
+
+Here is a relatively short guide on how to contribute code to Eddington. Please follow
+the next steps carefully when making a pull request.
+
+Step 1 – Write Your Code
+------------------------
+
+As of any open-source project in Github, contribution to Eddington is done via a
```
Comment (saroad2) [inline docs/contribution_guide.rst:14]: projects, not project

File: docs/contribution_guide.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+Code reviews you’ll be receiving often would address the following guidelines, as well
+as design issues if there are any.
+
+Step 2 – Test Your Code
+-----------------------
+
+In the Eddington platform, we believe in 100% test coverage, and we enforce it
+throughout our repositories! If you add new functionalities or change existing
+functionality, you must test the new ability with a unit test
+(or better yet – **tests**).
```
Comment (saroad2) [inline docs/contribution_guide.rst:33]: **unit tests**

File: docs/contribution_guide.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+3. This will run the unit tests and show the coverage report. Make sure that all the tests are passing and that each and every line of code is covered.
+
+Code coverage and extensive unit testing is something we will never compromise on.
+
+Step 3 – Clean Your Code
+------------------------
+
+Writing a working code can sometimes be really hard, but writing **clean** code is always
+harder. Here in the Eddington platform, we believe that code should be clean, but we
+want to make sure that you write clean code as easy as possible, using automatic tools
```
Comment (saroad2) [inline docs/contribution_guide.rst:52]: as easily

File: docs/contribution_guide.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+In order to use *Statue*, follow the next steps:
+
+1. Run :code:`pip install statue`
+2. Go to the main repository directory and run :code:`statue --context format`. This will change your code to fit styling guidelines. Save the changes in a commit or append them to an existing commit.
+3. Run :code:`statue` again (now without any arguments) and it will check if there are any issues that it wasn't able to solve on its own. If there are any errors, fix them.
+4. Save all changes in a commit or append them to an existing commit.
+
+You may find some of the errors presented by those tools tedious or irrelevant,
+but rest assured that we take those seriously. If you think that in a specific line
```
Comment (saroad2) [inline docs/contribution_guide.rst:67]: those errors

File: docs/contribution_guide.rst (hunk 1 above)
Comment (AssafZohar) [reply docs/contribution_guide.rst:14]: actually project is correct

File: docs/contribution_guide.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+Step 3 – Clean Your Code
+------------------------
+
+Writing a working code can sometimes be really hard, but writing **clean** code is always
+harder. Here in the Eddington platform, we believe that code should be clean, but we
+want to make sure that you write clean code as easily as possible, using automatic
+tools that help you achieve that along the way.
+
+We use state-of-the-art static code analysis tools such as black, flake8, pylint, mypy,
+pydostyle, etc. Orchestrating all these tools is statue_, which runs each of those
```
Comment (yuvalbardugo) [inline docs/contribution_guide.rst:56]: pydocstyle

**Comment by OhadNir9** [comment]
`statue --context format` from step 3 doesn't work for me. Did you mean `statue run --context format`?

**Comment by saroad2** [comment]
You're completely right. Fixed that in #59

**Pull request merged**